test/functional/tests/lazy_writes/test_lazy_writes_clean.py

```diff
@@ -0,0 +1,382 @@
+        description: |
+          Test if OpenCAS removes core from cache in modes with lazy writes and with different
+          filesystems without data loss.
+        pass_criteria:
+          - Core removing works properly.
+          - Data on core device is correct after core is removed.
+    """
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(256, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(512, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+        Udev.disable()
+
+    with TestRun.step(f"Start cache in {cache_mode} mode."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step(f"Add core with {fs.name} filesystem to cache and mount it."):
+        core_part.create_filesystem(fs)
+        core = cache.add_core(core_part)
+        core.mount(mnt_point)
+
+    with TestRun.step("Disable cleaning and sequential cutoff."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Create test file and read its md5 sum."):
+        test_file_main = create_random_test_file("/tmp/test_file_main", Size(64, Unit.MebiByte))
+        test_file_md5sum_main = test_file_main.md5sum()
+
+    with TestRun.step("Copy test file to the exported object."):
+        test_file_1 = File.create_file(mnt_point + "test_file_1")
+        dd = Dd().output(test_file_1.full_path) \
+            .input(test_file_main.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_1.refresh_item()
+        sync()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if test_file_md5sum_main != test_file_1.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Unmount and remove core."):
+        core.unmount()
+        core.remove_core()
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Mount core device."):
+        core_part.mount(mnt_point)
+
+    with TestRun.step("Read data from the core device."):
+        test_file_2 = File.create_file("/tmp/test_file_2")
+        dd = Dd().output(test_file_2.full_path) \
+            .input(test_file_1.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_1.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_2.refresh_item()
+        sync()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if test_file_md5sum_main != test_file_2.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+        if test_file_1.md5sum() != test_file_2.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
```

Comment:
unnecessary - you're checking earlier if test_file_md5sum_main equals test_file_1.md5sum()